fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersWriteServiceImpl.java

```diff
@@ -99,6 +101,8 @@ public CommandProcessingResult buybackLoanByLoanId(JsonCommand command) {
         validateBuybackRequestBody(command.json());
         Long loanId = command.getLoanId();
         validateLoan(loanId);
+        Integer loanStatus = fetchLoanStatus(loanId);
+        validateLoanStatus(loanStatus);
```

Comment:
It was not mistake to not checking the loan status for buyback! We dont wanna validate it!